### DISCUSSION TITLE
WinForms Designer - System.IO.FileNotFoundException when adding ChromiumWebBrowser to TabControl

### DISCUSSION THREAD

**Discussion by diman4ik** [cefsharp/CefSharp#4203] [General]
Design time error in winforms editor. When I put ChromiumWebBrowser on first tab of System.Windows.Forms.TabControl, than switch to second tab, than go back to first tab and get System.IO.FileNotFoundException in winforms designer. Visual Studio 2017, Visual Studio 2022 the same. This is exception message:
---------------------------
Microsoft Visual Studio
---------------------------
The control System.Windows.Forms.TabControl has thrown an unhandled exception in the designer and has been disabled.  

Exception:

Could not load file or assembly 'CefSharp.Core.Runtime, Version=104.4.180.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138' or one of its dependencies.

Stack trace:

   at System.Windows.Forms.TabControl.UpdateTabSelection(Boolean updateFocus)

   at System.Windows.Forms.TabControl.OnSelectedIndexChanged(EventArgs e)

   at System.Windows.Forms.TabControl.WmSelChange()

   at System.Windows.Forms.TabControl.WndProc(Message& m)

Project builds just fine, starts and works in debug and release modes as expected, problem is only in winforms designer. The problem is very easy to reproduce: just create Winforms app with single form,  than put TabControl on it, than put ChromiumWebBrowser on TabControl. If there is a fix how to solve it, I will be glad to know about it.

**Comment by amaitland** [comment]
There is no fix currently. You are welcome to debug and submit a PR. 

Unfortunately the stack trace doesn't have anything meaningful. Sometimes there's a more meaningful error in the event log. 

When in design mode we need to avoid loading any types that come from CefSharp.Core.Runtime.dll 

https://github.com/cefsharp/CefSharp/blob/cefsharp/104/CefSharp.WinForms/ChromiumWebBrowser.cs#L558
